### PR TITLE
JFG-992: replace boolean flags by lock

### PR DESCRIPTION
**Pull request by anton-antonenko**
Should be merged after https://github.com/griddynamics/jagger/pull/693

### CODE REVIEW

File: chassis/core/src/main/java/com/griddynamics/jagger/user/test/configurations/load/JLoadProfileUsers.java [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+
+    private long lifeTimeInSeconds;
+
+    private int startDelayInSeconds;
+
+    private int slewRateUsersPerSecond;
+
+    public JLoadProfileUsers(int numberOfUsers) {
+        this.numberOfUsers = numberOfUsers;
+        this.lifeTimeInSeconds = 60 * 60 * 48; // 2 days
```
Comment (SokolAndrey) [inline chassis/core/src/main/java/com/griddynamics/jagger/user/test/configurations/load/JLoadProfileUsers.java]: 60 * 60 * 24 * 2 maybe?

File: chassis/core/src/main/java/com/griddynamics/jagger/master/MasterToJaasCoordinator.java [hunk 2]
```diff
@@ -185,9 +188,7 @@ public boolean isRegistered() {
         
         private volatile RequestEntity<TestEnvironmentEntity> requestEntity;
 
-        private TestEnvironmentStatus lastStatus = TestEnvironmentStatus.PENDING;
-
-        private volatile boolean runningStatusSent = false;
+        private final Object statusLock = new Object();
```
Comment (SokolAndrey) [inline chassis/core/src/main/java/com/griddynamics/jagger/master/MasterToJaasCoordinator.java]: Maybe using _Lock_ is better than _synchronized_?

Comment (anton-antonenko) [inline chassis/core/src/main/java/com/griddynamics/jagger/master/MasterToJaasCoordinator.java]: For this case synchronized is better.
Performance of synchronized and ReentrantLock is the same, but ReentrantLock needs a try-catch-finally block which makes code dirtier, and synchronized doesn't need it.

File: chassis/core/src/main/java/com/griddynamics/jagger/master/MasterToJaasCoordinator.java [hunk 3]
```diff
@@ -91,6 +91,9 @@ public String awaitConfigToExecute() throws TerminateException {
                 }
             } while (configName == null);
             statusExchangeThread.setRunningRequestEntity(configName);
+            synchronized (statusExchangeThread.statusLock) {
+                statusExchangeThread.statusLock.wait();
```
Comment (a-badaev) [inline chassis/core/src/main/java/com/griddynamics/jagger/master/MasterToJaasCoordinator.java]: need to check 'standBy' periodically or can hang forever

File: chassis/core/src/main/java/com/griddynamics/jagger/util/generators/WorkloadGenerator.java [hunk 4]
```diff
@@ -19,6 +27,20 @@ static WorkloadClockConfiguration generateLoad(JLoadProfile jLoadProfile) {
             ((RpsClockConfiguration) clockConfiguration).setValue(((JLoadProfileRps) jLoadProfile).getRequestsPerSecond());
             ((RpsClockConfiguration) clockConfiguration).setWarmUpTime(((JLoadProfileRps) jLoadProfile).getWarmUpTimeInSeconds());
             ((RpsClockConfiguration) clockConfiguration).setMaxThreadNumber((int) ((JLoadProfileRps) jLoadProfile).getMaxLoadThreads());
+        } else if (jLoadProfile instanceof JLoadProfileUserGroups) {
```
Comment (a-badaev) [inline chassis/core/src/main/java/com/griddynamics/jagger/util/generators/WorkloadGenerator.java]: isn't it from another PR?

Comment (anton-antonenko) [inline chassis/core/src/main/java/com/griddynamics/jagger/util/generators/WorkloadGenerator.java]: yes, this PR must be merged after #693.
due to 'aux' I wasn't able to make any changes in code, that's why I had to push that changes also.
don't pay attention to it.

File: chassis/core/src/main/java/com/griddynamics/jagger/master/MasterToJaasCoordinator.java (hunk 3 above)
Comment (anton-antonenko) [inline chassis/core/src/main/java/com/griddynamics/jagger/master/MasterToJaasCoordinator.java]: do you mean this?

```
synchronized (statusExchangeThread.statusLock) {
    while (standBy)
        statusExchangeThread.statusLock.wait();
}
```

**Pull request merged**